helix-term/src/commands/typed.rs

```diff
@@ -322,6 +322,45 @@ fn force_write(
     write_impl(cx, args.first(), true)
 }
 
+fn rename_impl(cx: &mut compositor::Context, new_name: &str) -> anyhow::Result<()> {
+    let (_view, doc) = current!(cx.editor);
+
+    let path = if let Some(path) = doc.relative_path() {
+        path
+    } else {
+        bail!("File must exist to rename.");
+    };
+
+    let mut new_path = path.clone();
+    new_path.set_file_name(OsStr::new(new_name));
+
+    ensure!(
+        new_path.parent() == path.parent(),
+        "Root path must be the same."
+    );
+
+    ensure!(!new_path.exists(), "File already exists.");
```

Comment:
There could be a `:rename!`/`:rnm!` version that skips this check so that renaming can overwrite existing files

Reply:
Shouldn't `:rename` work only on a filename (no path), while `:move` would take a full path?